### PR TITLE
Improve logging, add logging tests for Lock service

### PR DESCRIPTION
**Pull request by hannesg**
This refactoring has become quite lengthy. I've switched the logging library to zap which has some nice perks when it comes to formatting and testing. Namely the formatter can now be switched to gcp optimized form.

Also, I added log tests to the Lock module. Once this MR is approved, I will add them to the remaining service as well.

### CODE REVIEW

File: chart/values.yaml.tpl [hunk 1]
```diff
@@ -8,6 +8,11 @@ git:
 
 hub: eu.gcr.io/freiheit-core
 tag: "$VERSION"
+log:
+  # Other possible values are "gcp" for a gcp-optimized format and "default" for json
```
Comment (sven-urbanski-freiheit-com) [inline chart/values.yaml.tpl:12]:
```suggestion
  # Possible values are "gcp" for a gcp-optimized format and "default" for json
```
(if there are actual "other" possible values, then please mention them in the comment)

File: chart/values.yaml.tpl [hunk 2]
```diff
@@ -8,6 +8,11 @@ git:
...
+  format: ""
+  # Other possible values are "DEBUG", "INFO", "ERROR"
```
Comment (sven-urbanski-freiheit-com) [inline chart/values.yaml.tpl:14]:
```suggestion
  # Possible values are "DEBUG", "INFO", "WARN", "ERROR"
```

File: services/cd-service/pkg/cmd/server.go [hunk 3]
```diff
@@ -62,106 +64,116 @@ func (c *Config) readPgpKeyRing() (openpgp.KeyRing, error) {
...
 
-	lockSrv := &service.LockServiceServer{
-		Repository: repo,
-	}
-	deploySrv := &service.DeployServiceServer{
-		Repository: repo,
-	}
+		repo, err := repository.New(ctx, repository.Config{
+			URL:            c.GitUrl,
+			Path:           "./repository",
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/cmd/server.go:88]: I think we should use c.repository here
(I know you didn't introduce it now :) )

File: services/cd-service/pkg/service/lock_test.go [hunk 4]
```diff
@@ -229,24 +238,90 @@ func TestInvalidCreateEnvironmentApplicationLockArguments(t *testing.T) {
...
+	}{
+		{
+			Name: "create an environment lock",
+			Request: &api.CreateEnvironmentApplicationLockRequest{
+				Environment: "bar",
+				Application: "app",
+				LockId:      "lock",
+			},
+			ExpectedCode:    codes.Internal,
+			ExpectedMessage: "internal error",
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/service/lock_test.go:281]: hm, could we provide a more meaningful error message here?
It's not clear to me, what is actually tested (or: why do we get this error here)

File: services/cd-service/pkg/cmd/server.go (hunk 3 above)
Comment (hannesg) [reply services/cd-service/pkg/cmd/server.go:88]: :+1: let's do that in a seperate MR

File: services/cd-service/pkg/service/lock_test.go (hunk 4 above)
Comment (hannesg) [reply services/cd-service/pkg/service/lock_test.go:281]: :thinking: reading this now, I think that shouldn't be tested in the test case at all.

File: services/frontend-service/Makefile [hunk 5]
```diff
@@ -38,7 +38,7 @@ bin/main: $(GO_FILES) | proto
 GARBAGE += bin/main
 
 run: bin/main
-	KUBERPULT_CDSERVER=localhost:8443 bin/main
+	KUBERPULT_CDSERVER=172.17.0.3:8443 bin/main
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/Makefile:41]: please don't commit this

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/frontend-service/Makefile (hunk 5 above)
Comment (hannesg) [reply services/frontend-service/Makefile]: Good point! I've made it possible to specify this on the command line

**Pull request merged**